services/QuillLMS/app/services/classlink_integration/district_importer.rb

```diff
@@ -62,7 +73,17 @@ def persist_records
         UserPersister.run(incoming_user:, existing_provider_users:, classlink_instance_id:)
       end
 
-      enrollments.each do |incoming_enrollment|
+      # these arrays must be refreshed, since the preceeding persisters may have created new records
```

Comment:
We could use the `users` array instead of making another DB call here. 

I favor the current implementation because:
1. `users` is an array of hashes that have not yet gone through the `User` active record validations. We don't want to throw errors on new Enrollments that reference a user that doesn't exist due to failed validation.
2. Using `users` would make the the `EnrollmentPersister` parameter shape subtly different from the other persisters, which may be confusing.
3. `DistrictImporter` is a nightly sync, so the small performance hit of re-querying the DB seems acceptable.